contracts/Commitment/TimedStateMachine.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity 0.4.15;
+
+import './StateMachine.sol';
+
+
+//  ------ time ----->
+//  +--------+-----------+--------+------------
+//  | Before | Whitelist | Public | Finished …
+//  +--------+-----------+--------+------------
+contract TimedStateMachine is StateMachine {
+
+    ////////////////////////
+    // Constants
+    ////////////////////////
+
+    int256 internal constant MIN_BEFORE_DURATION = 1 days;
+
+    int256 internal constant WHITELIST_DURATION = 5 days;
+
+    int256 internal constant PUBLIC_DURATION = 30 days;
+
+    //
+    // Starting time relative to WHITELIST_START
+    //
+
+    int256 internal constant PUBLIC_START = WHITELIST_DURATION;
```

Comment:
I would use a name that says that is it relative. Typically you expect timestamp from `XXXX_START` names

Reply:
:+1: